wasp-ai/src/client/pages/MainPage.jsx

```diff
@@ -68,8 +70,46 @@ const MainPage = () => {
 
   const [appAuthMethod, setAppAuthMethod] = useState(availableAuthMethods[0]);
 
+  useEffect(() => {
+    const appDetails = JSON.parse(localStorage.getItem("appDetails"));
```

Comment:
`JSON.parse` can often fail, let's put this in a try catch block with some default values